### ISSUE TITLE
Adding items on the same menu level in descending order (Possible bug)

### ISSUE THREAD

**Issue by travislbrundage** [adgsm/multi-level-push-menu#140]
Hi there,

So I am attempting to use the `additems` function in order to add new items to my menu. I want to add them in descending order. So, when clicking the [ add item button ], it adds a new item to the bottom of the list. I thought using the position field would allow me to do this, but it doesn't work as I expect.

Here is basically what I am doing:
```
title = "Main menu" // This is in the <h2> at the upper level of the menu
index = 0;
addItem = function() {
var $addTo = $('#menu').multilevelpushmenu('findmenusbytitle', title).first();
var item = [ { // My data here } ];
$('#menu').multilevelpushmenu('additems', item, $addTo, index);
index += 1;
}
```
I have affirmed that the menu grabbed by $addTo is the correct menu, and index is what I expect it to be. It correctly adds the first item, but afterwards it will embed all future items within the last item added. To make it more clear...

This is what is happening:
```
Main Menu
---------------
[ add item button ]
```
One click:
```
Main Menu
---------------
Item 1 >
[ add item button ]
```
Two clicks:
```
Main Menu
---------------
Item 1 >
[ add item button ]
```
Where is item 2? If I open Item 1...
```
Item 1
--------
Item 2 >
[ item data ]
```
The same thing will happen if I go back and add a third item. I will have to click into item 1, then into item 2 to see my item 3 embedded within it.

This is what I want to happen:
```
Main Menu
---------------
[ add item button]
```
One click:
```
Main Menu
---------------
Item 1 >
[ add item button ]
```
Two clicks:
```
Main Menu
---------------
Item 1 >
Item 2 >
[ add item button ]
```
And so on.

The closest I can get is if I just set position to be 0. Ergo, `$('#menu').multilevelpushmenu('additems', item, $addTo, 0);` instead. But this has the following result:
```
Main Menu
---------------
[ add item button ]
```
One click:
```
Main Menu
---------------
Item 1 >
[ add item button ]
```
Two clicks:
```
Main Menu
---------------
Item 2 >
Item 1 >
[ add item button ]
```
As you can see, the order of the items is opposite of what I want. I want it descending, not ascending.

Finally, notably, I can also set the position to 1 and it will work in a new way. `$('#menu').multilevelpushmenu('additems', item, $addTo, 1);` results in this:
```
Main Menu
---------------
[ add item button ]
```
One click:
```
Main Menu
---------------
[ add item button ]
Item 1 >
```
In this situation, `position = 1` placed it below my [ add item button ]. This makes sense, but earlier when I was iterating through, at index = 0, it added the item in the main menu as expected, but then at index = 1, it placed the next item within the menu of the first item.

Is this a bug with the `addItems` function? I'm still trying to step through the code, but looking at the javascript code in the library, it looks like it would function how I expect it to when iterating through. It just isn't for some reason.

**Comment by adgsm** [comment]
About the first issue... be careful when using 'findmenusbytitle' method since it will look for title(s) in the menu. If you have added same title many times it will look for the first one and add next item(s) within it as a submenu. I'm just guessing it since you didn't expose what your items are looking like and I suppose you use a loop for adding them... 

> var item = [ { // My data here } ];

Since $addTo is actually representing div element (jQuery) where items are to be injected I suggest you to directly reference to dom object where you want items to be added. For example you can find currently expanded element with
```
/**
 * Active (last expanded) menu level
 * Returns active menu level object, or false if menu is fully collapsed
 */
$('#menu').multilevelpushmenu('activemenu');
```

and then use it as $addTo.

Second issue is really not an issue since you always add items to position 0 within items array.

Bottom line, make sure you're using correct $addTo div element and all the rest should be ok...

**Comment by travislbrundage** [comment]
Hi adgsm,

Thanks for the prompt reply.

Sorry, that was my mistake on $addTo - I actually have it like that currently: `var $addTo = $('#menu').multilevelpushmenu('activemenu').first();`. The button for adding a new item is in the top level of my menu, so it should add a new item in the top level of the menu upon clicking.

I'm not sure I made the issue clear. Everything "works" but I want to add items to the menu in descending order, not ascending. Here's more precise code:
```
index = 1;
addItem = function() {
var $addTo = $('#menu').multilevelpushmenu('findmenusbytitle', title).first();
var item = [
        {
          name: 'Chapter ' + index,
          link: '#',
          items: [
            {
              title: 'Chapter ' + index,
              icon: 'fa fa-bookmark',
              items: [
                {
                  name: 'Chapter Info',
                  icon: 'fa fa-info-circle',
                  link: '#'
                },
                {
                  name: 'Delete Chapter',
                  id: 'deleteChapter',
                  icon: 'fa fa-trash-o',
                  link: '#'
                }
              ]
            }
          ]
        }
      ];
$('#menu').multilevelpushmenu('additems', item, $addTo, 0);
index += 1;
}
```

This gives me the following result:
```
Main Menu
---------------
[ add item button ]
```
One click:
```
Main Menu
---------------
Item 1 >
[ add item button ]
```
Two clicks:
```
Main Menu
---------------
Item 2 >
Item 1 >
[ add item button ]
```

I want `Item 1 >` to appear on top, and `Item 2 >` to appear below it. Then, if I add another `Item 3 >`, I want it to appear below `Item 2>`, and so on. How can I make this happen? I thought it was by changing position to something other than 0, but it resulted in the embedded items as I described originally.

**Comment by adgsm** [comment]
If you have mistyped
`$('#menu').multilevelpushmenu('additems', item, $addTo, 0); `
instead of
`$('#menu').multilevelpushmenu('additems', item, $addTo, index);`
than this is a bug, possibly on line
https://github.com/adgsm/multi-level-push-menu/blob/master/jquery.multilevelpushmenu.js#L345
where it should be
`( position < ( $itemGroup.children( 'li' ).length ) && position >= 0 ) ?  `
instead of
`( position < ( $itemGroup.find( 'li' ).length ) && position >= 0 ) ?  `

Simple workaround for you could be to set index to -1,as follows
`$('#menu').multilevelpushmenu('additems', item, $addTo , -1 );`
which will then always trigger $.appendTo and as so add to the end of the queue.

Thanks for spotting this...

**Comment by travislbrundage** [comment]
Excellent, this fixed the problem! Thanks @adgsm !
